### ISSUE TITLE
MaplyQuadImageTilesLayer + MaplyMBTileSource sometimes shows a blackscreen

### ISSUE THREAD

**Issue by alasarr** [mousebird/WhirlyGlobe#248]
Sometimes (5% more or less) when I try to load a MBTile in WhirlyGlobe I get a black screen, neither errors nor exceptions are thrown, just a black screen.  I'm using WhirlyGlobe 2.2. 

It looks like an issue with my app but I don't know how to fix/debug it. I think this because I've tried a very simple project (just a WhirlyGlobe without any other library), using the same set of MBTiles and it works fine, no black screens anymore.

Could it be a compile param in my app project? Do you have experimented similar issues?

Any ideas that help me here would be really appreciated.

Here the code

Add to globe
``` 
MaplyMBTileSource *tileSource = [[MaplyMBTileSource alloc] initWithMBTiles:self.resource];
        
    layer = [[MaplyQuadImageTilesLayer alloc] initWithCoordSystem:tileSource.coordSys tileSource:tileSource];
    layer.handleEdges = true;
    layer.coverPoles = true;
    layer.importanceScale = 3;
    layer.maxTiles = 512;
    [self.globe addLayer:layer];
    layer.drawPriority = self.zIndex;
```

Remove from globe
```
   [self.globe removeLayer:layer];
```

**Comment by alasarr** [comment]
After several debugging with instruments I've found a memory leak with the sqlite3 inside MaplyMBTileSource that could be related with this bug. It looks like the sqlite connection was not close and a new allocation happened each time a new MaplyMBTileSource object was created.

Once I've fixed it I'm not able to reproduce this issue again, so I think If you approved this pull-request https://github.com/mousebird/WhirlyGlobe/pull/281 we can close it.


I hope this helps!
